ddtrace/settings/http.py

```diff
@@ -37,8 +38,11 @@ def trace_headers(self, whitelist):
                 continue
             self._whitelist_headers.add(normalized_header_name)
 
+        self.header_is_traced.invalidate()
+
         return self
 
+    @cachedmethod()
```

Comment:
aside: is it worth having an `if not self._whitelist_headers: return False` to this method?

would save the cost of normalizing the header, but this whole method being cached means we only need to normalize the header once on the first time we see it.

? /shrug